torchao/quantization/_transform_module.py

```diff
@@ -0,0 +1,17 @@
+from typing import Callable, Dict
+
+import torch
+
+from torchao.core.config import AOBaseWorkflowConfig
+
+_QUANTIZE_CONFIG_HANDLER: Dict[
```

Comment:
alternative we could use `singledispatch` do this: _QUANTIZE_CONFIG_HANDLER: Dict[
    AOBaseWorkflowConfig,
    Callable[[torch.nn.Module, AOBaseWorkflowConfig], torch.nn.Module],
] = {}


def register_quantize_module_handler(config_type):
    def decorator(func):
        _QUANTIZE_CONFIG_HANDLER[config_type] = func

    return decorator
    
    
    which would make quantize_ generic and then we register instantations baed off of config type

Reply:
if the first argument of `quantize_` was the configuration, I would see the benefit in using `singledispatch`.  The first argument is currently the `model` though.